src/app/complex/complex-results/complex-navigator/table-structure/table-interactor-column/table-interactor-column.component.html

```diff
@@ -0,0 +1,97 @@
+<div class="Complex-navigator">
+  <table class="interactors-table"
+         [style.--cols]="complexes.length"
+         [style]="complexes.length>8 ? 'width: calc(130px + 70px * var(--cols) + 183px)':''">
+    <ng-container *ngFor=" let interactor of enrichedInteractors; let i=index">
+      <tr *ngIf="!interactor.hidden">
+        <ng-container *ngFor="let oneType of ranges" class="interactorsOrdering">
+          <td *ngIf="oneType[2]===i" [attr.rowspan]="oneType[1]" class="interactorSeparation"
+              title="{{oneType[0]}}">
+            <div class="interactorNameContainer" [style.--rowspanSize]="oneType[1]">
+              {{ oneType[0] }}
+            </div>
+          </td>
+        </ng-container>
+        <td class="interactorsColumn" [ngClass]="{horizontalHeader : complexes.length<=8}"
+            [style]="ranges.length!=0 ? 'left:3ch':''">
+          <cp-table-interactor-name
+            [interactorName]="interactor.interactor.name"
+            [interactorType]="interactor.interactor.interactorType"
+            [interactorId]="interactor.interactor.identifier"
+            [interactorOrganism]="interactor.interactor.organismName"
+            [identifierLink]="interactor.interactor.identifierLink"
+            [organismIconDisplay]="organismIconDisplay"
+            [interactorTypeDisplay]="interactorTypeDisplay"
+            [IDDisplay]="IDDisplay"
+          ></cp-table-interactor-name>
+          <!-- Icon for the expandable -->
+          <div class="container" *ngIf="interactor.isSubComplex">
+            <a title="{{ interactor.interactor.name }} interactors" (click)="toggleSubcomplexExpandable(i)">
+              Interactors
+              <i class="icon icon-common" data-icon="&#xf078;"
+                 style="font-size:small;
+                "></i>
+            </a>
+          </div>
+        </td>
+        <!-- Interactors' stoichiometry -->
+        <ng-container *ngFor="let complex of enrichedComplexes">
+          <td class="intStoich">
+            <cp-table-main-interactor
+              [complex]="complex"
+              [i]="i"
+              [enrichedInteractors]="enrichedInteractors"
+            ></cp-table-main-interactor>
+          </td>
+        </ng-container>
+        <div class="spaceHolder" *ngIf="complexes.length>8"></div>
+      </tr>
+      <!-- Expandable menu for subcomplexes -->
+      <ng-container *ngIf="interactor.expanded">
+        <ng-container *ngIf="!!interactor.subComponents">
+          <tr *ngFor="let el of interactor.subComponents; let j=index"
+              class="expandedRows"
+              [ngClass]="{'firstExpandedRow': j === 0, 'lastExpandedRow': j === interactor.subComponents.length - 1}">
```

Comment:
The issue with the cells not having the right width with a low number of complexes is in this line. I'm still looking into it, but this is definitely the line.

Reply:
I think I got it.

In `table-interactor-column.component.css`, for class `interactorsColumn`, get rid of `width: 130px;` (line 16); and for class `intStoich`, get rid of `width: 70px;` (line 32).

Reply:
I think I've put it this way because of the interactors header which was not sizing the right way but I will check that tomorrow because it was working well on the previous versions

Reply:
I managed to fix this issue.

1. You may have some CSS changes locally, revert them as what you have here is not far from working.

2. Update classes in table-interactor-column.component.css
    ```
    .interactorsColumn {
      position: sticky;
      left: 0;
      z-index: 3;
      max-height: 45px;
      min-height: 45px;
      min-width: 130px;
      max-width: 130px;
      padding: 5px;
      text-align: end;
      color: #0e6f76;
      font-weight: bold;
      background: inherit;
    }
    
    .interactorsColumnFixedWidth {
      width: 130px;
    }
    
    .intStoich {
      height: 45px;
      max-width: 70px;
      min-width: 70px;
      font-size: medium;
      text-align: end;
      z-index: 1;
    }

    .intStoichFixedWidth {
      width: 70px;
    }
    ```

3. Update html to use new CSS classes
    3.1. lines 15-16
    ```
    <td class="interactorsColumn" [ngClass]="complexes.length<=8 ? 'horizontalHeader' : 'interactorsColumnFixedWidth'"
            [style]="ranges.length!=0 ? 'left:3ch':''">
    ```
    3.2. line 39
    ````
    <td class="intStoich" [ngClass]="{'intStoichFixedWidth': complexes.length > 8}">
    ````
    3.3. line 82
    ```
    <td class="intStoich" [ngClass]="{'intStoichFixedWidth': complexes.length > 8}">
    ```
    3.4. also line 91, not classes but there is a bug there
    ```
    <div class="spaceHolder" *ngIf="complexes.length>8"></div>
    ```

Reply:
I tested it with the searches
- nuclear pore
- KMN: 9 results
- CPX-4441: 3 results